File: config/supabaseEnv.ts
export const SUPABASE_URL =
    process.env.SUPABASE_URL || process.env.SUPABASE_PROJECT_URL;

export const SUPABASE_ANON_KEY =
    process.env.SUPABASE_ANON_KEY || process.env.SUPABASE_ANON_KEY;

export const SUPABASE_SERVICE_ROLE_KEY =
    process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.SUPABASE_SERVICE_ROLE_KEY;

// For legacy code paths expecting a generic key:
export const SUPABASE_KEY =
    process.env.SUPABASE_KEY || SUPABASE_SERVICE_ROLE_KEY || SUPABASE_ANON_KEY;

if (!SUPABASE_URL || !SUPABASE_KEY) {
    throw new Error("SUPABASE_URL or SUPABASE_KEY missing/invalid");
}

File: config/supabase_env.py
import os
from dataclasses import dataclass


@dataclass(frozen=True)
class SupabaseEnv:
    url: str | None
    publishable_key: str | None
    secret_key: str | None


def load_supabase_env() -> SupabaseEnv:
    url = os.getenv("SUPABASE_URL") or os.getenv("SUPABASE_PROJECT_URL")

    publishable_key = (
        os.getenv("SUPABASE_ANON_KEY")
        or os.getenv("SUPABASE_ANON_KEY")  # legacy fallback
    )

    secret_key = (
        os.getenv("SUPABASE_SERVICE_ROLE_KEY")
        or os.getenv("SUPABASE_SERVICE_ROLE_KEY")  # legacy fallback
    )

    return SupabaseEnv(url=url, publishable_key=publishable_key, secret_key=secret_key)
